tests/lib/openstack_query/managers/test_query_manager.py

```diff
@@ -86,32 +90,62 @@ def test_build_and_run_query_with_runner_params(
             ("with both", MOCKED_PRESET_DETAILS, MOCKED_OUTPUT_DETAILS),
         ]
     )
-    @patch("openstack_query.managers.query_manager.QueryManager._populate_query")
-    @patch("openstack_query.managers.query_manager.QueryManager._get_query_output")
+    @patch("openstack_query.managers.query_manager.QueryOutputDetails")
     def test_build_and_run_query(
         self,
         _,
         mock_preset_details,
         mock_output_details,
-        mock_get_query_output,
-        mock_populate_query,
+        mock_query_output_details,
     ):
         """
         Tests that _build_and_run_query method functions expectedly
         Sets up a QueryResource object and runs a given query with appropriate inputs (no runner params).
         Returns query result
         """
-        res = self.instance._build_and_run_query(
-            mock_output_details, mock_preset_details
-        )
 
-        mock_populate_query.assert_called_once_with(
-            preset_details=mock_preset_details,
-            properties_to_select=mock_output_details.properties_to_select,
-        )
+        if not mock_output_details:
+            # just assume valid output details is made from calling from_kwargs
+            mock_query_output_details.from_kwargs.return_value = MOCKED_OUTPUT_DETAILS
+
+        with patch(
+            "openstack_query.managers.query_manager.QueryManager._populate_output_params"
+        ) as mock_populate_output_params:
+            with patch(
+                "openstack_query.managers.query_manager.QueryManager._get_query_output"
```

Comment:
This many patches for a single method call is a very strong indicator it's doing too much

The class should break these down into public API steps with a top level one eg.

```
class QueryManager:

   # args omitted for brevity
    def build_query():
    def execute_query():
    def output_query():

    def run_full_query():
        built = self.build_query()
        result = self.execute_query(built)
        return self.output_query(result)
```

Since it's just as valid for a unit test, or an external caller to call self.output_query even if there's a helper method to stitch it up.

Our tests then simply test each step in isolation and test that the top level method calls all of those methods which we replace with mocks in `test_run_full_query`

Reply:
moving this into separate PR - issue #103